NEWS.md

```diff
@@ -66,6 +66,9 @@ Command-line option changes
 Multi-threading changes
 -----------------------
 
+* A new `:dynamic` schedule option for `Threads.@threads` which is similar to the default behavior except iterations
+  will be scheduled dynamically to available worker threads rather than pinned to each thread. This option is more
+  composable if existing `@spawn`-ed tasks are taking place, and can be nested inside other `@threads` loops ([#43919])
```

Comment:
`@threads for` was already allowed so I think "can be nested' can be confusing. Maybe simply

```suggestion
  composable with (possibly nested) `@spawn` and `@threads` loops ([#43919])
```